cmd/grafana-agent-flow: add agentctl tools (#4208)
grafana/agent: docs/sources/flow/reference/cli/tools.md

---
title: grafana-agent tools
weight: 100
---

# `grafana-agent tools` command

The `grafana-agent tools` command contains command line tooling grouped by Flow
component.

> > **NOTE**: utilities in this command have no backwards compatibility
> > guarantees and may change or be removed between releases.

Reviewer: ```suggestion
{{% admonition type="note" %}}
Utilities in this command have no backward compatibility
guarantees and may change or be removed between releases.
{{% /admonition %}}
```

## Subcommands

### prometheus.remote_write sample-stats

Usage: `grafana-agent tools prometheus.remote_write sample-stats [FLAG ...] WAL_DIRECTORY`

The `sample-stats` command reads the Write-Ahead Log (WAL) specified by
`WAL_DIRECTORY` and collects information on metric samples within it.

For each metric discovered, `sample-stats` emits:

* The timestamp of the oldest sample received for that metric.
* The timestamp of the newest sample received for that metric.
* The total number of samples discovered for that metric.

By default, `sample-stats` will return information for every metric in the WAL.

> The `--selector` flag can optionally be passed to filter the reported metrics
> to a smaller set.

Reviewer: ```suggestion
You can pass the `--selector` flag to filter the reported metrics to a smaller set.
```

> The following flags are supported:

Reviewer: Is this part of the template style? Since it's one flag, it should be `The following flag is supported:` but I can see the argument for a template style usecase here

Author: I don't think we necessarily have a defined style yet, so I'll change it to make it clearer that it's just a single flag 👍

* `--selector`: A PromQL label selector to filter data by. (default `{}`)

### prometheus.remote_write target-stats

Usage: `grafana-agent tools prometheus.remote_write target-stats [FLAG ...] WAL_DIRECTORY`

The `target-stats` command reads the Write-Ahead Log (WAL) specified by
`WAL_DIRECTORY` and collects metric cardinality information for a specific
target.

For the target specified by the `--job` and `--instance` flags, unique metric
names for that target are printed along with the number of series with that
metric name.

The following flags are supported:

* `--job`: The `job` label of the target.
* `--instance`: The `instance` label of the target.

> The `--job` and `--instance` labels are required.

Reviewer 2: If these are required should they be defined in the `Usage:` above for copy pastability?

### prometheus.remote_write wal-stats

Usage: `grafana-agent tools prometheus.remote_write wal-stats WAL_DIRECTORY`

The `wal-stats` command reads the Write-Ahead Log (WAL) specified by

> `WAL_DIRECTORY` and collects general information about the it.

Reviewer: ```suggestion
`WAL_DIRECTORY` and collects general information about it.
```
